extmod/network_ifx_whd.c

```diff
@@ -0,0 +1,634 @@
+        if (bssid.len != 6) {
+            if (strncmp(bssid.buf, (char *)whd_bssid.octet, 6)) {
+                mp_raise_ValueError(NULL);
+            }
+        }
+    }
+
+    // Extract the security type, if given.
+    whd_security_t auth_type;
+    if (args[ARG_security].u_int == -1) {
+        if (key.buf == NULL || key.len == 0) {
+            auth_type = WHD_SECURITY_OPEN;
+        } else {
+            auth_type = WHD_SECURITY_WPA2_MIXED_PSK;
+        }
+    } else {
+        auth_type = args[ARG_security].u_int;
+    }
+
+    // Extract channel, if given.
+    if (args[ARG_channel].u_int == -1) {
+
+        uint32_t ret = whd_wifi_set_channel(itf, args[ARG_channel].u_int);
+        whd_assert_raise(ret);
+    }
+
+    // Start the WiFi join procedure.  It will run in the background.
+    uint32_t ret = whd_wifi_join(itf, &whd_ssid, auth_type, key.buf, key.len);
+    if (ret != WHD_SUCCESS) {
+        mp_raise_OSError(-ret);
+    }
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_KW(network_ifx_whd_connect_obj, 1, network_ifx_whd_connect);
+
+STATIC mp_obj_t network_ifx_whd_disconnect(mp_obj_t self_in) {
+    network_ifx_whd_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    whd_interface_t itf = *(self->itf);
+    uint32_t ret = whd_wifi_leave(itf);
+    whd_assert_raise(ret);
+
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(network_ifx_whd_disconnect_obj, network_ifx_whd_disconnect);
+
+STATIC mp_obj_t network_ifx_whd_isconnected(mp_obj_t self_in) {
+    network_ifx_whd_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    whd_interface_t itf = *(self->itf);
+    // Is this function the most suitable? There is no whd public
+    // function to get the link status
+    return mp_obj_new_bool(whd_wifi_is_ready_to_transceive(itf));
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(network_ifx_whd_isconnected_obj, network_ifx_whd_isconnected);
+
+STATIC mp_obj_t network_ifx_whd_ifconfig(size_t n_args, const mp_obj_t *args) {
+    network_ifx_whd_obj_t *self = MP_OBJ_TO_PTR(args[0]);
+    return mod_network_nic_ifconfig(&(self->netif), n_args - 1, args + 1);
+}
+MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(network_ifx_whd_ifconfig_obj, 1, 2, network_ifx_whd_ifconfig);
+
+STATIC mp_obj_t network_ifx_whd_status(size_t n_args, const mp_obj_t *args) {
+    network_ifx_whd_obj_t *self = MP_OBJ_TO_PTR(args[0]);
+    whd_interface_t itf = *(self->itf);
+
+    if (n_args == 1) {
+        // no arguments: return link status
+        return MP_OBJ_NEW_SMALL_INT(whd_wifi_is_ready_to_transceive(itf));
+    }
+
+    // one argument: return status based on query parameter
+    switch (mp_obj_str_get_qstr(args[1])) {
+        case MP_QSTR_stations: {
+            // return list of connected stations
+            // if (self->itf != CYW43_ITF_AP) {
+            //     mp_raise_ValueError(MP_ERROR_TEXT("AP required"));
+            // }
+            int num_stas;
+            uint8_t macs[32 * 6];
+            // cyw43_wifi_ap_get_stas(self->cyw, &num_stas, macs);
```

Comment:
![image](https://user-images.githubusercontent.com/105282041/224336055-572be384-2d35-4e0d-ad3c-67a833a13736.png)

A warning is observed in the build. Maybe we might not solve this right now since this relates to AP mode. 

But just to keep in mind for later. Such as equivalent function as in line 384 must be implemented.